build_artifacts/v2/v2.2/v2.2.0/dirs/etc/inference-server/server.py

```diff
@@ -0,0 +1,101 @@
+
+
+class InferenceServer(TornadoServer):
+    def __init__(self):
+        self._environment = Environment()
+        logger.setLevel(self._environment.logging_level)
+        logger.debug(f"Environment: {str(self._environment)}")
+
+        self._path_to_inference_code = (
+            Path(self._environment.base_directory).joinpath(self._environment.code_directory)
+            if self._environment.code_directory else
+            Path(self._environment.base_directory)
+        )
+        logger.debug(f"Path to inference code: `{str(self._path_to_inference_code)}`")
+
+        self._handler = None
+
+    def initialize(self):
+        self._install_runtime_requirements()
+        self._handler = self._load_inference_handler()
+
+    def _install_runtime_requirements(self):
+        logger.info("Installing runtime requirements...")
+        
+        requirements_txt = self._path_to_inference_code.joinpath(self._environment.requirements)
+        if requirements_txt.is_file():
+            try:
+                subprocess.check_call(
+                    [sys.executable, "-m", "pip", "install", "-r", str(requirements_txt)]
+                )
+            except Exception as e:
+                raise RequirementsInstallException(e)
+        else:
+            logger.debug(f"No requirements file was found at `{str(requirements_txt)}`")
+
+    def _load_inference_handler(self) -> callable:
+        logger.info("Loading inference handler...")
+
+        inference_module_name, handle_name = self._environment.code.split(".")
+        if inference_module_name and handle_name:
+            inference_module_file = f"{inference_module_name}.py"
+            module_spec = importlib.util.spec_from_file_location(
+                inference_module_file,
+                str(self._path_to_inference_code.joinpath(inference_module_file))
+            )
+            if module_spec:
+                sys.path.insert(0, str(self._path_to_inference_code.resolve()))
+                inference_module = module_spec.loader.load_module(inference_module_file)
+                if hasattr(inference_module, handle_name):
+                    handler = getattr(inference_module, handle_name)
+                else:
+                    raise InferenceCodeLoadException(
+                        f"Handler `{handle_name}` could not be found in module `{inference_module_file}`"
+                    )
+                logger.debug(f"Loaded handler `{handle_name}` from module `{inference_module_name}`")
+                return handler
+            else:
+                raise InferenceCodeLoadException(
+                    f"Inference code could not be found at `{str(self._path_to_inference_code.joinpath(inference_module_file))}`"
+                )
+        raise InferenceCodeLoadException(
+            f"Inference code expected in the format of `<module>.<handler>` but was provided as {code}"
+        )
+
+
+if __name__ == "__main__":
+    inference_server = InferenceServer()
```

Comment:
Nit: add some more logs in initialize and serve

Reply:
ack will do